build.cake

```diff
@@ -116,20 +118,37 @@ Task("Publish")
     .WithCriteria(BuildSystem.IsRunningOnTeamCity)
     .Does(() =>
 {
-    NuGetPush($"{artifactsDir}Octostache.{nugetVersion}.nupkg", new NuGetPushSettings {
-        Source = "https://f.feedz.io/octopus-deploy/dependencies/nuget",
-        ApiKey = EnvironmentVariable("FeedzIoApiKey"),
-        SkipDuplicate = true,
+    var currentBranch = GitBranchCurrent(DirectoryPath.FromString(".")).FriendlyName;
+    var octopusServer = EnvironmentVariable("OctopusServerUrl");
+    var octopusApiKey = EnvironmentVariable("OctopusServerApiKey");
+    var space = EnvironmentVariable("OctopusServerSpaceName");
+    var octopusProjectName = "Octostache";
+
+    var nugetPackage = GetFiles($"{artifactsDir}Octostache.{nugetVersion}.nupkg");
+
+    // Current config for this repo doesn't generate prerelease tags, even if we're on a development/feature branch.
+    // Using the --overwrite-mode=IgnoreIfExists flag instructs the target Octopus server to ignore any attempted uploads with the same verison number.
+    // This decision is made server-side, so you'll still see log messages indicating the package was uploaded. Never fear, the push is discarded if the version already exists. 
+    // You can verify this by looking at the SHA1 and Published date of the package in the package feed: it won't change on subsequent pushes.
+    OctoPush(octopusServer, octopusApiKey, nugetPackage, new OctopusPushSettings 
+    {
+        ArgumentCustomization = args => args.Append("--overwrite-mode=IgnoreIfExists"),
+        Space = space 
     });
 
-    if (gitVersionInfo.PreReleaseTagWithDash == "")
-    {
-        NuGetPush($"{artifactsDir}Octostache.{nugetVersion}.nupkg", new NuGetPushSettings {
-            Source = "https://www.nuget.org/api/v2/package",
-            ApiKey = EnvironmentVariable("NuGetApiKey"),
-            SkipDuplicate = true,
-        });
-    }
+    // Config-as-Code doesn't yet support Automatic Release Creation, so do it manually
+    OctoCreateRelease(octopusProjectName, new CreateReleaseSettings {
```

Comment:
I would recommend doing away with cake during this uplift. my branch is here - https://github.com/OctopusDeploy/Octostache/compare/ankith/fm/nuke?expand=1 it's still a WIP but should serve as a starting point

Reply:
Not going to do this right now, we might get to it later.